### ISSUE TITLE
Moving to Openlayers 5

### ISSUE THREAD

**Issue by darkscript** [Viglino/ol-ext#123]
We have some legacy openlayers animation [here](https://github.com/Viglino/ol-ext/blob/52ed071d1e05b690d47bce3692b3f5e74ac93ebf/src/control/Globe.js#L173) , [ also here](https://github.com/Viglino/ol-ext/blob/52ed071d1e05b690d47bce3692b3f5e74ac93ebf/src/control/Overview.js#L152), [here](https://github.com/Viglino/ol-ext/blob/52ed071d1e05b690d47bce3692b3f5e74ac93ebf/src/control/Overview.js#L159)

**Comment by Viglino** [comment]
Yes its old ol version (<3.20).
I think we can remove it for good.
I'll try tout push it this week.
@+

**Comment by darkscript** [comment]
Yeah, or maybe we can update them to [view.animate](http://openlayers.org/en/latest/apidoc/ol.View.html#animate) in current openlayers

**Comment by Viglino** [comment]
The view.animate is already used. There is just a fallback when it doesn't exit (older version). We just have to remove the test.

I can see you're working on an ol-5 branch. Be aware we have to provided a full build (./dist) for those who use it. This means we have to 

I think we can still use: `import ol_control_Control from 'ol/control/Control'`. The name is internal to the file and doesn't have any effect on the rest of the code, but we can easily convert it to `ol.control.Control`.
If you use `import Control from 'ol/control/control'` we can't guest it is an `ol.control.Control`.
According to [this](https://github.com/openlayers/openlayers/wiki/OpenLayers-v5.x#users-of-the-v4x-ol-package) we just have to verify the CamelCase rule. The default exports of v4.x will still work though.
````javascript
import ol_Map from 'ol/Map';

export function createMap() {
  return new ol_Map({});
}
````
Do you have a repo as [ol-ol-ext-webpack-example](https://github.com/darkscript/ol-ol-ext-webpack-example) with to work on it.
@+

**Comment by darkscript** [comment]
Oh, I never saw the guide, also I was wrong on using ``import Control from 'ol/control/control'`` which will make it impossible to generate the dist ol-ext version. Maybe I can lay low abit and wait for official ol-5 release. What do you think?

**Comment by Viglino** [comment]
I think we can test the pre-release [v5.0.0-beta.10](https://github.com/openlayers/openlayers/releases/tag/v5.0.0-beta.10) to see if all is running well and what is broken.
I've test a bit and it seems that some default exports are missing. I don't know if it will be the case in the final release (but it seems to be).
For example, `ol` is no more exported and we need to import the `inherits` method:
````javascript
import ol from 'ol'
...
ol.inherits(ol_control_Search, ol_control_Control);
````
should be: 
````javascript
import {inherits as ol_inherits} from 'ol'
...
ol_inherits(ol_control_Search, ol_control_Control);
````
I've the same issue with all ol namespaces: `ol.proj`, `ol.coordinate`, etc.
````javascript
import {add as ol_coordinate_add} from 'ol/coordinate'
import {transform as ol_proj_transform} from 'ol/proj'
...
ol_coordinate_add(coord, 5);
var c = ol_proj_transform(coord, 'EPSG:4326', 'EPSG:4358');
````
All other classes (with CamelCase) should work properly with no modification as they are exported as default (I guess).

I don't know how it will run with jsDoc and I propose to stay with the classical namespace (ol.coordinate) still we get an other idea :smile:.

**Comment by darkscript** [comment]
Yeah you are right, I think the only things to change is ``inherits`` , ``ol.proj`` and ``ol.coordinate`` . Maybe you can do a global refactoring of the imports and we can have a **ol-ext-2.10-beta** . Maybe you can do a new branch and I will try searching for errors.

**Comment by Viglino** [comment]
I've already done some test on your [example repo](https://github.com/darkscript/ol-ol-ext-webpack-example). I can remove the error on the ol-ext lib but I've still get some warning on ol itself.
The Swipe control doesn't work properly but I don't know if it is the pre-release or not.

**Comment by Viglino** [comment]
Hello,
I've been working on restoring the example with the last ol version (now v5!).
The main issue was on deprecated functionalities and the removal of the optional opt_this argument on Observable: 
````javascript
obs.on('change', fn, this)
````
is now 
````javascript
obs.on('change, fn.bind(this));
````
So far all is working with v4.6.5 and v5.0.0 (with the dist) excepted the `ol.control.Profil` that is using an ol.Sphere that is no more available in ol5  (use ol/sphere functions instead) and all examples are running.

Now I'll been working on moving to ol5 but this will introduce breaking changes with the ol4
I've created a new branch ol-5 to handle changes (main changes concerned inherits , ol.proj, ol.coordinate and ol/sphere + CamelCase modules).

@+

**Comment by darkscript** [comment]
This is great work @Viglino , I will create a test app with ol-ext@v2.0.5 and ol5 to do more tests. Will later upgrade my main projects to ol5. Thanks much.

**Comment by Viglino** [comment]
Use the ol-5 branch to test.
The last commit includes a first test to have the src/control/Profile.js working (https://github.com/Viglino/ol-ext/commit/fc702b9d51863b3153db3fbda78462c98a722dce#diff-3cce39e052bcb5bd5955f32dbe81c470).
You can see the main import changes:
````javascript
import {inherits as ol_inherits} from 'ol'
import {getDistance as ol_sphere_getDistance} from 'ol/sphere'
import {transform as ol_proj_transform} from 'ol/proj'

import ol_control_Control from 'ol/control/Control'
import ol_Feature from 'ol/Feature'
````

**Comment by darkscript** [comment]
@Viglino in the ol-migration, all refactoring is done and I think all migration is done except for ol_coordinate including [GeomUtils](https://github.com/Viglino/ol-ext/blob/ol5-migration/geom/GeomUtils.js). You can maybe work on that, otherwise there should be few errors and we can fix as we go

**Comment by Viglino** [comment]
I've done some tests and it seems ok to me.
I'll try to publish a new  npm version (2.0.6) for use with ol<5 tomorrow.
Then we can merge the branch to master and publish a v3 to use with ol5.

**Comment by Viglino** [comment]
Hello,
I've publish a new v2.0.6 on npm.
@darkscript you can merge your branch to the master to prepare a new version (v3)  to use with ol5 :smiley: 
@+

**Comment by darkscript** [comment]
Hi,
@Viglino I merged ol-migration with master, did some refactoring to [**ol_coordinate**](https://github.com/Viglino/ol-ext/commit/88223135b0f75adcdcf2d45e2fbbc68bebdad412) and this will probably break some of your examples, I didn't generate **dist** folder

**Comment by Viglino** [comment]
OK I'll have a look!

**Comment by Viglino** [comment]
I've build the dist and the examples are running (except [snapguide](file:///C:/wamp64/www/github/ol-ext/examples/interaction/map.interaction.snapguides.html).
I'll try to investigate later...

**Comment by Viglino** [comment]
All examples are running :smile: 
I'll try to test modules and I think we can publish a new npm version.

**Comment by Viglino** [comment]
New version v3.0.0 is now available on npm ol-ext@3.0.0
